src/BasisMatrices.jl

```diff
@@ -70,7 +70,8 @@ export BasisFamily, Cheb, Lin, Spline, Basis,
 # functions
 export nodes, get_coefs, funfitxy, funfitf, funeval, evalbase,
        derivative_op, row_kron, evaluate, fit!, update_coefs!,
-       complete_polynomial, complete_polynomial!, n_complete
+       complete_polynomial, complete_polynomial!, n_complete,
+       complete_polynomial_der, complete_polynomial_der!
```

Comment:
Great point.

I don't think we can use a keyword argument for derivatives (it wouldn't dispatch on it) -- I think the following protocol seems sensible

```
# computes 3rd order complete polynomial for z
complete_polynomial(z, 3)

# computes derivatives with respect to variable 1 of 3rd order complete polynomial
complete_polynomial(z, 3, 1)
```

Thoughts?